mrmap/resourceNew/models/mapcontext.py

```diff
@@ -46,11 +47,22 @@ class MapContextLayer(MPTTModel):
                              blank=True,
                              verbose_name=_("title"),
                              help_text=_("a short descriptive title for this map context layer"))
-
-    # todo referenz auf Dataset (mit Layer)
-    # todo referenz auf Layer
-    # todo referenz auf FeatureType (zukünftig)
-    # zukünftig: featuretype, kml, gml, ...
+    dataset_metadata = models.ForeignKey(DatasetMetadata,
+                                         on_delete=models.CASCADE,
+                                         null=True,
+                                         blank=True)
+    # TODO
+    # layer_metadata = models.ForeignKey(LayerMetadata,
+    #                                   on_delete=models.CASCADE,
+    #                                   null=True,
+    #                                   blank=True)
+    # TODO
+    #feature_type_metadata = models.ForeignKey(FeatureTypeMetadata,
+    #                                          on_delete=models.CASCADE,
+    #                                          null=True,
+    #                                          blank=True)
+    preview_image = models.ImageField(null=True, blank=True)
```

Comment:
upload_to argument is unused. This will result in messy upload structure. See example below for better usage:

```python
def xml_backup_file_path(instance, filename):
    # file will be uploaded to MEDIA_ROOT/xml_documents/<id>/<filename>
    return 'xml_documents/{0}/{1}'.format(instance.pk, filename)


class DocumentModelMixin(models.Model):
    """Abstract model to store documents such as capabilities or ISO MD_Metadata xml for different related objects.

    :attr xml_backup: the original xml as backup to restore the xml fields.
    """
    xml_mapper_cls = None
    xml_backup_file = models.FileField(verbose_name=_("xml backup"),
                                       help_text=_("the original xml as backup to restore the xml field."),
                                       upload_to=xml_backup_file_path,
                                       editable=False)

```